### PR TITLE
Allow passing sparse matrices to _parse_n_wd()

### PR DESCRIPTION
**Pull request by vmarkovtsev**

### CODE REVIEW

File: python/artm/batches_utils.py [hunk 1]
```diff
@@ -232,23 +237,33 @@ def __reset_batch():
...
-                    item.token_id.append(batch_vocab[token])
-                    item.token_weight.append(float(value))
+            if not isinstance(column, np.ndarray):
+                if isinstance(column, spmatrix):
+                    nnz = column.nonzero()[1]
+                    enum = zip(nnz, np.squeeze(column[0, nnz].toarray()))
+                elif isinstance(column, np.matrix):
+                    enum = enumerate(np.squeeze(np.asarray(column)))
+                else:
+                    raise TypeError("Unsupported column type: %s" % type(column))
```
Comment (MelLain) [inline python/artm/batches_utils.py]: Please use "{}".format() instead of "%s"

File: python/artm/batches_utils.py [hunk 2]
```diff
@@ -223,6 +223,11 @@ def __reset_batch():
             batch.id = str(uuid.uuid4())
```
Comment (MelLain) [inline python/artm/batches_utils.py]: This change should be covered by new test in this file

https://github.com/bigartm/bigartm/blob/master/python/tests/artm/test_batches_utils.py

**Comment by vmarkovtsev** [comment]
@MelLain done. I had to refactor the tests.

`scipy` is not required by the package, but it is required to run the tests.

**Comment by MelLain** [comment]
@vmarkovtsev Please revert back the test_func() function and put your code as one more part of it (as it done in other tests). Though your structure is much more better, all tests code should be similar for now.

Such a refactoring will be done, but it requires:
- changing all files together
- creating file with common functions outside of tests (creating temp folder, some checks, asserts etc.)

**Comment by vmarkovtsev** [comment]
@MelLain I would rape my interns for such a bad tests organization, but as you wish.

**Comment by MelLain** [comment]
I created a special ticket for tests code (https://github.com/bigartm/bigartm/issues/834).

Don't forget to rebase your change (
git fetch upstream
git rebase upstream/master
)
upstream == bigartm/bigartm.git

**Comment by vmarkovtsev** [comment]
@MelLain `test_dump_load.py` should be fixed, it is [flaky](https://testing.googleblog.com/2016/05/flaky-tests-at-google-and-how-we.html)

**Comment by vmarkovtsev** [comment]
@MelLain All is done on my side

**Comment by MelLain** [comment]
Great!

**Comment by vmarkovtsev** [comment]
What?!

**Comment by vmarkovtsev** [comment]
@MelLain You forged my author identity and discarded my authorship. Please revert my commit and let me do the proper contribution.

**Comment by MelLain** [comment]
Oh, I'm very sorry, I've forgotten that merge operation will override the authorship info:(

Feel free to make your contribution, but please change the constant from flaky test to 0.005, that it hasn't any large meaning but will prevent troubles with tests.

**Pull request closed without merging**